### ISSUE TITLE
finetune_regressor.py in the examples. I have a few questions

### ISSUE THREAD

**Issue by jutleo** [PriorLabs/TabPFN#577]
### 1、In the code:
```
regressor.fit_from_preprocessed(
    X_trains_preprocessed, y_trains_znorm, cat_ixs, confs
)
```
Why is X here the preprocessed data while y is the z-normalized data? Could there be a mistake here?

### 2、In this part:
```
mse, mae, r2 = evaluate_regressor(
    regressor, eval_config, X_train, y_train, X_test, y_test
)
```
It makes sense to use raw data for evaluation, but looking at the overall process, something feels off.

I still can't fully understand the complete fine-tuning process and can't pinpoint the exact issue. The observable problem is that the MSE and MAE values are increasing.

**Comment by iivalchev** [comment]
Hey @jutleo, a bit of reasoning behind the regressor fine-tuning can be found here: https://medium.com/@iivalchev/how-to-fine-tune-tabpfn-on-your-data-a831b328b6c0

To your questions:

1. It is not a mistake, as the preprocessing of the data is done "outside" of the regressor the `get_preprocessed_datasets` on the regressor doesn't modify the regressor it just pre-processes the data so that you can feed to a DataLoader. Think of `get_preprocessed_datasets` as detaching the pre-processing (of course based on the config set on the regressor). So in the training loop you need to pass the preprocessed data back to the regressor in order to do a forward pass. This time however you are passing tensors directly. In case of regression the `y` must be z-normalized.

2. You are fine-tuning the transformer, the preprocessing is not changed. So in order to be sure that your evaluation will match inference you create a new regressor with preprocessing not detached with the fine-tuned transformer. This regressor needs the actual unprepossessed data. This is safety precaution (IMO).

To your last point, yes fine-tuning can be tricky. You need be careful and experiment. I think I mentioned that either here or on Discord, but you need to tweak lr, as well as how large is your training and test set in the forward pass as this kinda forms a batch on which gradients will be calculated. Also for regression I have noticed the norms of the gradient to become quite large which leads to unstable training. I suspect this is due to the discretization done by bar-distribution and in particular the small width of the center bars. One more resource that can you help you with the training loop is here: https://github.com/LennartPurucker/finetune_tabpfn_v2/blob/master/finetuning_scripts/finetune_tabpfn_main.py it has examples of graint clipping and accumulation.

**Comment by oscarkey** [comment]
hi @jutleo. I'm closing this issue for now, but feel free to re-open if you need further help :)
